test/e2e/e2e_toomanyrestarts_test.go

```diff
@@ -0,0 +1,188 @@
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Name:            "pause",
+						ImagePullPolicy: "Always",
+						Image:           "kubernetes/pause",
+						Command:         []string{"/bin/sh"},
+						Args:            []string{"-c", "sleep 1s && exit 1"},
+						Ports:           []v1.ContainerPort{{ContainerPort: 80}},
+					}},
+				},
+			},
+		},
+	}
+	_, err = clientSet.AppsV1().Deployments(deploymentObj.Namespace).Create(ctx, deploymentObj, metav1.CreateOptions{})
+	if err != nil {
+		t.Logf("Error creating deployment: %v", err)
+		if err = clientSet.AppsV1().Deployments(deploymentObj.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set(map[string]string{"test": "restart-pod", "name": "test-toomanyrestarts"})).String(),
+		}); err != nil {
+			t.Fatalf("Unable to delete deployment: %v", err)
+		}
+		return
+	}
+	defer clientSet.AppsV1().Deployments(deploymentObj.Namespace).Delete(ctx, deploymentObj.Name, metav1.DeleteOptions{})
+
+	// Need to wait restartCount more than 4
+	result, err := waitPodRestartCount(ctx, clientSet, testNamespace.Name, t)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if !result {
+		t.Fatal("Pod restart count not as expected")
+	}
+
+	tests := []struct {
+		name                    string
+		podRestartThreshold     int32
+		includingInitContainers bool
+		expectedEvictedPodCount int
+	}{
+		{
+			name:                    "test-no-evictions",
+			podRestartThreshold:     int32(10000),
+			includingInitContainers: true,
+			expectedEvictedPodCount: 0,
+		},
+		{
+			name:                    "test-one-evictions",
+			podRestartThreshold:     int32(4),
+			includingInitContainers: true,
+			expectedEvictedPodCount: 4,
+		},
+	}
+	for _, tc := range tests {
+		evictionPolicyGroupVersion, err := eutils.SupportEviction(clientSet)
+		if err != nil || len(evictionPolicyGroupVersion) == 0 {
+			t.Fatalf("Error creating eviction policy group: %v", err)
+		}
+		podEvictor := evictions.NewPodEvictor(
+			clientSet,
+			evictionPolicyGroupVersion,
+			false,
+			0,
+			nodes,
+			true,
+			false,
+			false,
+		)
+		// Run RemovePodsHavingTooManyRestarts strategy
+		t.Log("Running RemovePodsHavingTooManyRestarts strategy")
+		strategies.RemovePodsHavingTooManyRestarts(
+			ctx,
+			clientSet,
+			deschedulerapi.DeschedulerStrategy{
+				Enabled: true,
+				Params: &deschedulerapi.StrategyParameters{
+					PodsHavingTooManyRestarts: &deschedulerapi.PodsHavingTooManyRestarts{
+						PodRestartThreshold:     tc.podRestartThreshold,
+						IncludingInitContainers: tc.includingInitContainers,
+					},
+				},
+			},
+			workerNodes,
+			podEvictor,
+		)
+
+		waitForTerminatingPodsToDisappear(ctx, t, clientSet, testNamespace.Name)
+		actualEvictedPodCount := podEvictor.TotalEvicted()
+		if actualEvictedPodCount < tc.expectedEvictedPodCount {
```

Comment:
Let's use "!=" to report even the case when more pods than expected are evicted.

Reply:
Sounds reasonable, thanks for the clarification.